rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -1492,63 +1598,176 @@ void game_list_frame::ShowContextMenu(const QPoint &pos)
 			CreateCPUCaches(gameinfo);
 		}
 	});
-	connect(remove_game, &QAction::triggered, this, [this, current_game, gameinfo, cache_base_dir, name]
+	connect(remove_game, &QAction::triggered, this, [this, current_game, gameinfo, cache_base_dir, hdd1_cache_base_dir, name]
 	{
-		if (current_game.path.empty())
+		if (is_game_running(current_game.serial))
 		{
-			game_list_log.fatal("Cannot remove game. Path is empty");
+			QMessageBox::critical(this, tr("Cannot Remove Game"), tr("The PS3 application is still running, it cannot be removed!"));
 			return;
 		}
 
-		if (is_game_running(current_game.serial))
+		QString text;
+		const bool disc_avail = qstr(current_game.category) == cat::cat_disc_game;
+		const bool disc_auto_detect = disc_avail && (current_game.path.starts_with(rpcs3::utils::get_games_dir()));
+		const std::string data_sub_dir = GetFirstSubDirBySerial(rpcs3::utils::get_hdd0_dir() + "/game/", current_game.serial);
+
+		// If data path is present (it could be absent for a disc game)
+		const bool data_avail = !disc_avail || data_sub_dir != "";
+		// Data path to be removed (if any)
+		const std::string data_dir = !disc_avail ? current_game.path : (data_avail ? rpcs3::utils::get_hdd0_dir() + "/game/" + data_sub_dir : "");
+
+		text = tr("%0 - %1\n").arg(qstr(current_game.serial)).arg(name);
```

Comment:
Move the QString text declaration from above to here
```suggestion
		QString text = tr("%0 - %1\n").arg(qstr(current_game.serial)).arg(name);
```